diff --git a/dist/esm/index.d.ts b/dist/esm/index.d.ts
index 72c5cb4ded2bedfb57ab91c0c28571a7211b1cbe..2bed773e8d853d991810769ac2918b8c4fd1b3f0 100644
--- a/dist/esm/index.d.ts
+++ b/dist/esm/index.d.ts
@@ -69,7 +69,11 @@ declare namespace Umami {
             value: number;
             change: number;
         };
-        uniques: {
+        visitors: {
+            value: number;
+            change: number;
+        };
+        visits: {
             value: number;
             change: number;
         };
@@ -77,7 +81,7 @@ declare namespace Umami {
             value: number;
             change: number;
         };
-        totalTime: {
+        totaltime: {
             value: number;
             change: number;
         };
@@ -290,7 +294,7 @@ declare class UmamiApiClient {
     getWebsiteStats(websiteId: string, params: {
         startAt: number;
         endAt: number;
-        url: string;
+        url?: string;
         referrer?: string;
         title?: string;
         query?: string;
